src/profile-logic/profile-data.js

```diff
@@ -3371,6 +3375,136 @@ export function findAddressProofForFile(
   return null;
 }
 
+/**
+ * Calculate a lower bound for the function size, in bytes, of a native symbol.
+ * This is used when the symbol server does not return a size for a function.
+ * We need to know the size when we want to show assembly code for the function,
+ * in order to know how many bytes to disassemble.
+ * We estimate the size by finding the highest known address for this symbol in
+ * the frame table, and adding one byte (because the instruction at that address
+ * is at least one byte long).
+ */
+export function calculateFunctionSizeLowerBound(
+  frameTable: FrameTable,
+  nativeSymbolAddress: Address,
+  nativeSymbolIndex: IndexIntoNativeSymbolTable
+): Bytes {
+  let maxFrameAddress = nativeSymbolAddress;
+  for (let i = 0; i < frameTable.length; i++) {
+    if (frameTable.nativeSymbol[i] === nativeSymbolIndex) {
+      const frameAddress = frameTable.address[i];
+      if (frameAddress > maxFrameAddress) {
+        maxFrameAddress = frameAddress;
+      }
+    }
+  }
+  return maxFrameAddress + 1 - nativeSymbolAddress;
+}
+
+/**
+ * Gathers the native symbols for a given call node. In most cases, a call node
+ * just has one native symbol (or zero if it's not native code). But in some
+ * cases, a call node can have its native code in multiple different functions,
+ * for example if it was inlined into multiple different functions.
+ * Another way to get multiple native symbols here is if a call tree transform
+ * collapsed frames from different original functions into a merged function.
```

Comment:
Is the inverted call tree another example (like mentioned elsewhere)?

Reply:
Yes, I'll adjust the sentence about "inlined into multiple different functions" to mention the inverted tree.